sound/soc/sof/topology.c

```diff
@@ -1723,6 +1801,21 @@ static int sof_widget_load_pga(struct snd_soc_component *scomp, int index,
 		goto err;
 	}
 
+	ret = sof_parse_tokens(scomp, &volume->comp, core_tokens,
+			       ARRAY_SIZE(core_tokens), private->array,
+			       le32_to_cpu(private->size));
+	if (ret != 0) {
+		dev_err(scomp->dev, "error: parse volume.comp tokens failed %d\n",
+			ret);
+		goto err;
+	}
+
+	ret = sof_core_enable(sdev, volume->comp.core);
+	if (ret < 0) {
+		dev_err(scomp->dev, "error: enable core: %d\n", ret);
+		goto err;
+	}
```

Comment:
@lyakh I am wondering if it makes sense to maybe move the core_tokens parsing to sof_widget_ready() and pass the core_id as a parameter to all the widget load functions? Then, we could avoid adding it so many places

Reply:
@ranj063 this violates the current approach of parsing all tokens within each load function separately, but I agree, this would save a lot of code and consolidate the processing in one place, will change, thanks!